test/integration/spec/room.js

```diff
@@ -42,27 +43,71 @@ describe('Room', function() {
   // eslint-disable-next-line no-invalid-this
   this.timeout(60000);
 
-  (defaults.topology === 'peer-to-peer' ? describe.skip : describe)('.isRecording', () => {
-    [true, false].forEach(enableRecording => {
-      context(`when recording is ${enableRecording ? 'enabled' : 'disabled'}`, () => {
-        let room;
-        let sid;
+  (defaults.topology === 'peer-to-peer' ? describe.skip : describe)('recording', () => {
+    [true, false].forEach(trackSharedAtConnect => {
+      [true, false].forEach(recordingEnabledAtCreate => {
+        context(`when recording is ${recordingEnabledAtCreate ? 'enabled' : 'disabled'} and tracks ${trackSharedAtConnect ? 'published' : 'not published'} at connect `, () => {
+          let room;
+          let sid;
+          let localTrack;
+          const initialRecording = trackSharedAtConnect ? recordingEnabledAtCreate : false;
+          before(async () => {
+            sid = await createRoom(randomName(), defaults.topology, { RecordParticipantsOnConnect: recordingEnabledAtCreate });
+            localTrack = await waitFor(createLocalVideoTrack(), 'alice local video track');
```

Comment:
Please publish 2 tracks and verify `recordingStarted` and `recordingStopped` are fired only once each.